### PR TITLE
Adding dependency modules message

### PR DESCRIPTION
**Pull request by SatishRanjan**
Adding proto buff message definitions that will be sent from host to language worker for the language worker to download the dependent libraries.

### CODE REVIEW

File: src/proto/FunctionRpc.proto [hunk 1]
```diff
@@ -370,3 +371,8 @@ message RpcHttp {
   bool enable_content_negotiation= 16;
   TypedData rawBody = 17;
 }
+
+//Used to send the dependency management request to the language worker
+message DependencyManagementRequest {
```
Comment (peterhuene) [inline src/proto/FunctionRpc.proto]: Is this supposed to be part of `StreamingMessage.content`?

Comment (SatishRanjan) [reply src/proto/FunctionRpc.proto]: That's right @peterhuene , I'll add it to StreamingMessage

File: src/proto/Dependency.proto [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+option java_outer_classname = "DependencyProto";
+option csharp_namespace = "Microsoft.Azure.WebJobs.Script.Grpc.Messages";
+option go_package ="github.com/Azure/azure-functions-go-worker/internal/rpc";
+
+package AzureFunctionsRpcMessages;
+
+
+//Used to send the managed dependency module for the language worker to download the client module
+message Dependency {
+  string name = 1;
```
Comment (pragnagopa) [inline src/proto/Dependency.proto]: Add comments for each the fields explaining their purpose.

File: src/proto/Dependency.proto [hunk 3]
```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+// protobuf vscode extension: https://marketplace.visualstudio.com/items?itemName=zxh404.vscode-proto3
+
+option java_multiple_files = true;
+option java_package = "com.microsoft.azure.functions.rpc.messages";
+option java_outer_classname = "DependencyProto";
```
Comment (pragnagopa) [inline src/proto/Dependency.proto]: remove java and go. When the language worker is ready to consume this feature language worker owner will add this.

File: src/proto/FunctionRpc.proto [hunk 4]
```diff
@@ -10,6 +10,7 @@ option go_package ="github.com/Azure/azure-functions-go-worker/internal/rpc";
 package AzureFunctionsRpcMessages;
 
 import "google/protobuf/duration.proto";
+import "Dependency.proto";
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: Lets put these in a new folder "definitions" as @kashimiz suggested.

File: src/proto/FunctionRpc.proto [hunk 5]
```diff
@@ -69,6 +70,9 @@ message StreamingMessage {
     FunctionEnvironmentReloadRequest function_environment_reload_request = 25;
 
     FunctionEnvironmentReloadResponse function_environment_reload_response = 26;
+	
+	//Used to send the dependency management request to the language worker
+	DependencyManagementRequest dependency_management_request = 27;
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: nit: format

File: src/proto/FunctionRpc.proto [hunk 6]
```diff
@@ -370,3 +374,12 @@ message RpcHttp {
   bool enable_content_negotiation= 16;
   TypedData rawBody = 17;
 }
+
+// Language worker sends dependency download message back to host
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: Nit: Language worker sends dependency download status to the host

File: src/proto/FunctionRpc.proto [hunk 7]
```diff
@@ -370,3 +374,12 @@ message RpcHttp {
...
+message DependencyDownloadResponse {
+  // unique function identifier
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: You need to add this to the streaming message

File: src/proto/FunctionRpc.proto [hunk 8]
```diff
@@ -69,6 +70,9 @@ message StreamingMessage {
     FunctionEnvironmentReloadRequest function_environment_reload_request = 25;
 
     FunctionEnvironmentReloadResponse function_environment_reload_response = 26;
+	
+	//Used to send the dependency management request to the language worker
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: It is optional for a language worker to implement a new message type. Functions host needs to look for capabilities published by a language worker before sending this request.

File: src/proto/FunctionRpc.proto (hunk 1 above)
Comment (pragnagopa) [reply src/proto/FunctionRpc.proto]: @peterhuene - It is optional for a language worker to implement new message types. Functions host will check for capabilities sent in WorkerInitResponse  before sending this message.

File: src/proto/FunctionRpc.proto (hunk 4 above)
Comment (pragnagopa) [reply src/proto/FunctionRpc.proto]: Following the conventions similar to https://github.com/googleapis/googleapis/tree/master/google/bytestream, move new definition to its own folder

**Comment by pragnagopa** [comment]
@SatishRanjan - Please address feedback here and consume changes in the host repo

File: src/proto/Dependency.proto [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+package AzureFunctionsRpcMessages;
+
+
+//Used to send the managed dependency module for the language worker to download the client module
+message Dependency {
+  string name = 1;
+  string major_version = 2; 
+  string path = 3;
+  bool is_downloadrequired = 4;
```
Comment (peterhuene) [inline src/proto/Dependency.proto]:
```suggestion
  bool is_download_required = 4;
```

File: src/proto/ManagedDependency/Dependency.proto [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+package AzureFunctionsRpcMessages;
+
+
+//Used to send the managed dependency module for the language worker to download the client module
+message Dependency {
+  string name = 1;
+  string major_version = 2; 
+  string path = 3;
+  bool is_downloadrequired = 4;
```
Comment (peterhuene) [inline src/proto/ManagedDependency/Dependency.proto]: Same as previous, but this is the new location I assume?

```suggestion
  bool is_download_required = 4;
```

File: src/proto/ManagedDependency/Dependency.proto [hunk 11]
```diff
@@ -10,18 +10,23 @@ option go_package ="github.com/Azure/azure-functions-go-worker/internal/rpc";
 package AzureFunctionsRpcMessages;
 
 
-//Used to send the managed dependency module for the language worker to download the client module
+//Used to send the manaed dependency module for the language worker to download the client module
```
Comment (pragnagopa) [inline src/proto/ManagedDependency/Dependency.proto]: nit: typo managed

**Pull request merged**